### DISCUSSION TITLE
`enhance.image_histogram2d` compatibility with `torch.histogram`

### DISCUSSION THREAD

**Discussion by kunaltyagi** [kornia/kornia#2681] [Questions]
I'm wondering how I can get numerical outputs as close to `torch.histrogram` as possible (for single channel, single batch tensor).

![image](https://github.com/kornia/kornia/assets/2657068/407f0d7f-bdc6-44e3-b5c1-ef195f4178c2)

**Comment by edgarriba** [comment]
not quite sure but i think this two functions have different purposes. Image histogram sums** over the last two dimensions of an image. The torch one, might not consider any spatial relationship in the image ?

** https://kornia.readthedocs.io/en/latest/_modules/kornia/enhance/histogram.html#image_histogram2d

**Reply by kunaltyagi** [reply]
I'm using a single channel, single batch tensor. In this scenario, I want to know what settings to use to have the same outputs from both

**Reply by edgarriba** [reply]
i don't have an answer for that. You will have to check yourself torch implementation and see how it differ from ours

**Reply by kunaltyagi** [reply]
I found one issue in kornia's implementation: double counting due to `torch.abs(u) <= 1` (items on the boundary are counted twice)

**Reply by edgarriba** [reply]
Can you submit a PR including/updating tests?

**Reply by kunaltyagi** [reply]
I'll try to get that done sometime next week

**Reply by edgarriba** [reply]
awesome !
